src/components/radio-group/radio-group.ts

```diff
@@ -26,6 +26,26 @@ export default class SlRadioGroup extends LitElement {
   /** Shows the fieldset and legend that surrounds the radio group. */
   @property({ type: Boolean, attribute: 'fieldset' }) fieldset = false;
 
+  /** Indicates that a selection is required. */
+  @property({ type: Boolean, reflect: true }) required = false;
+
+  constructor() {
+    super();
+    this.addEventListener('sl-change', this.syncRadioButtons);
```

Comment:
This should probably run in `connectedCallback()`. If the element gets removed and added back to the DOM, for example, `disconnectedCallback()` will fire and remove the listener. The constructor only runs on creation, whereas `connectedCallback()` runs on connect and reconnect.